ports/rp2/modmachine.c

```diff
@@ -190,22 +193,17 @@ static void mp_machine_lightsleep(size_t n_args, const mp_obj_t *args) {
     // Disable ROSC.
     rosc_hw->ctrl = ROSC_CTRL_ENABLE_VALUE_DISABLE << ROSC_CTRL_ENABLE_LSB;
 
+    // pico-sdk alarm pool uses 3
+    // micropython soft timers use 2
+    int alarm_num = 1;
```

Comment:
When we moved away from the alarm pool the first time we had some internal discussion about trying to keep some alarms free for third party use. But, honestly doing it this way seems so much simpler and better - no more messing about with the possibility that the timer IRQ is disabled on the other core.

What's here is already a big improvement over magic number 3 in the original version, but suggest adding a constant like `MICROPY_HW_LIGHTSLEEP_ALARM_NUM` near mpconfigport.h:178 (which should make the extra comments redundant, too, as we can use that header as the source.)